drivers/media/i2c/mira220.c

```diff
@@ -0,0 +1,2087 @@
+	ret = mira220_otp_read(mira220, 0x19, 1, &val);
+	dev_err(&client->dev, "Read OTP add 0x19+1 with val %x\n", val);
+
+	mira220_otp_power_off(mira220);
+
+	return 0;
+}
+
+static const struct v4l2_subdev_core_ops mira220_core_ops = {
+	.subscribe_event = v4l2_ctrl_subdev_subscribe_event,
+	.unsubscribe_event = v4l2_event_subdev_unsubscribe,
+};
+
+static const struct v4l2_subdev_video_ops mira220_video_ops = {
+	.s_stream = mira220_set_stream,
+};
+
+static const struct v4l2_subdev_pad_ops mira220_pad_ops = {
+	.enum_mbus_code = mira220_enum_mbus_code,
+	.get_fmt = mira220_get_pad_format,
+	.set_fmt = mira220_set_pad_format,
+	.get_selection = mira220_get_selection,
+	.enum_frame_size = mira220_enum_frame_size,
+};
+
+static const struct v4l2_subdev_ops mira220_subdev_ops = {
+	.core = &mira220_core_ops,
+	.video = &mira220_video_ops,
+	.pad = &mira220_pad_ops,
+};
+
+static const struct v4l2_subdev_internal_ops mira220_internal_ops = {
+	.open = mira220_open,
+};
+
+/* Initialize control handlers */
+static int mira220_init_controls(struct mira220 *mira220)
+{
+	struct i2c_client *client = v4l2_get_subdevdata(&mira220->sd);
+	struct v4l2_ctrl_handler *ctrl_hdlr;
+	struct v4l2_fwnode_device_properties props;
+	int ret;
+
+	u32 max_exposure = 0;
+
+	ctrl_hdlr = &mira220->ctrl_handler;
+	/* v4l2_ctrl_handler_init gives a hint/guess of the number of v4l2_ctrl_new */
+	ret = v4l2_ctrl_handler_init(ctrl_hdlr, 16);
+	if (ret)
+		return ret;
+
+	mutex_init(&mira220->mutex);
+	ctrl_hdlr->lock = &mira220->mutex;
+	/* By default, PIXEL_RATE is read only */
+	mira220->pixel_rate = v4l2_ctrl_new_std(ctrl_hdlr, &mira220_ctrl_ops,
+						V4L2_CID_PIXEL_RATE,
+						mira220->mode->pixel_rate,
+						mira220->mode->pixel_rate, 1,
+						mira220->mode->pixel_rate);
+
+	mira220->vblank = v4l2_ctrl_new_std(ctrl_hdlr, &mira220_ctrl_ops,
+					    V4L2_CID_VBLANK,
+					    mira220->mode->min_vblank,
+					    mira220->mode->max_vblank, 1,
+					    mira220->mode->min_vblank);
+
+	mira220->hblank =
+		v4l2_ctrl_new_std(ctrl_hdlr, &mira220_ctrl_ops, V4L2_CID_HBLANK,
+				  mira220->mode->hblank, mira220->mode->hblank,
+				  1, mira220->mode->hblank);
+
+	// Exposure is indicated in number of lines here
+	// Max is determined by vblank + vsize and Tglob.
+	max_exposure = mira220_calculate_max_exposure_time(
+		mira220->mode->height, mira220->vblank->val,
+		mira220->mode->row_length);
+
+	mira220->exposure =
+		v4l2_ctrl_new_std(ctrl_hdlr, &mira220_ctrl_ops,
+				  V4L2_CID_EXPOSURE, MIRA220_EXPOSURE_MIN,
+				  max_exposure, 1, MIRA220_DEFAULT_EXPOSURE);
+
+	mira220->gain = v4l2_ctrl_new_std(
+		ctrl_hdlr, &mira220_ctrl_ops, V4L2_CID_ANALOGUE_GAIN,
+		MIRA220_ANALOG_GAIN_MIN, MIRA220_ANALOG_GAIN_MAX,
+		MIRA220_ANALOG_GAIN_STEP, MIRA220_ANALOG_GAIN_DEFAULT);
+
+	mira220->hflip = v4l2_ctrl_new_std(ctrl_hdlr, &mira220_ctrl_ops,
+					   V4L2_CID_HFLIP, 0, 1, 1, 0);
+	if (mira220->hflip)
+		mira220->hflip->flags |= V4L2_CTRL_FLAG_MODIFY_LAYOUT;
```

Comment:
In what way does it modify the layout? Normally this is when it changes the Bayer order, but your sensor appears not to do that.